core/model/impl/src/main/kotlin/ConditionalHoldingModelImpl.kt

```diff
@@ -70,20 +70,24 @@ internal open class ConditionalHoldingModelImpl(
 
         override val conditionScope: ConditionScope by lazy {
             featureTypes.fold(ConditionScope.Always as ConditionScope) { acc, featureType ->
-                acc and featureType.conditionScope
+                acc and (featureType.conditionScope ?: ConditionScope.Always)
             }
         }
 
         override fun validate(validator: Validator) {
             onlyIn.forEach(validator::child)
             featureTypes.forEach(validator::child)
 
-            when {
-                conditionScope.isContradiction() -> {
-                    validator.reportWarning(Strings.Warnings.contradictionCondition(conditionScope))
-                }
-                featureTypes.isNotEmpty() && conditionScope.isTautology() -> {
-                    validator.reportWarning(Strings.Warnings.tautologyCondition(conditionScope))
+            // If some feature is not valid, no need to report induced errors
+            if (featureTypes.none { it.conditionScope == null }) {
```

Comment:
This is a change that got `invalid-features-variants.golden` changed - we don't analyze scopes if some of them are invalid - this produces induced errors, that are of interest to nobody.